check_internal_errors

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
```

Comment:
This might be better added to the `hooks` directory.

Can you add a call to this to `hooks/pre-commit` ?  It shouldn't take long to run, so it's a good check to have there.